### ISSUE TITLE
IndentityId get refreshed after one hour, provoking 401 code response

### ISSUE THREAD

**Issue by yktoledo** [aws-amplify/amplify-android#2751]
### Before opening, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-android/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-android/discussions).

### Language and Async Model

Kotlin - Coroutines

### Amplify Categories

Authentication, REST API

### Gradle script dependencies

<details>

```groovy
// Put output below this line
    implementation 'com.amplifyframework:core:2.14.11'
    implementation 'com.amplifyframework:aws-api:2.14.11'
    implementation 'com.amplifyframework:aws-auth-cognito:2.14.11'

```

</details>

### Environment information

<details>

```
# Put output below this line

------------------------------------------------------------
Gradle 8.2
------------------------------------------------------------

Kotlin:       1.8.20
Groovy:       3.0.17
Ant:          Apache Ant(TM) version 1.10.13 compiled on January 4 2023
JVM:          17.0.7 (JetBrains s.r.o. 17.0.7+7-b985.2)
OS:           Mac OS X 14.4.1 aarch64


```

</details>

### Please include any relevant guides or documentation you're referencing

https://docs.amplify.aws/android/build-a-backend/auth/enable-guest-access/

### Describe the bug

I want to consume API Gateway resource using `authorizationType:AWS_IAM` and "Enable guest access" (Allow unauthenticated logins), based on the [documentation](https://docs.amplify.aws/android/build-a-backend/auth/enable-guest-access/#:~:text=you%20can%20obtain,method),  "you can obtain the unique Identity ID which is assigned to the device through the fetchAuthSession method", with the intention to use this "identity id" as the device identifier to communicate, after a little a bit more than one hour of fresh install,  the credentials get refreshed, which is perfect, but the identityId too, I will assume is not the expected behavior. 

Maybe I'm missing something, server configuration specific or android implementation detail, despite is so minimal the amount of code use, 4 line of code, add plugins call with api and auth, follow by the configure call, after that just the get call.

Following the doc https://docs.amplify.aws/android/build-a-backend/restapi/fetch-data/

```
// add this to log
Amplify.addPlugin(AndroidLoggingPlugin(LogLevel.VERBOSE))

Amplify.addPlugin(AWSApiPlugin())
Amplify.addPlugin(AWSCognitoAuthPlugin())
Amplify.configure(applicationContext)

val response = Amplify.API.get(request)
```

I appreciate it if you can help me on this.

Note: following the same steps in iOS is not having the same problem, iOS keeps the identity id all the time.

### Code Snippet

```java
// Put your code below this line.

```

### Log output

<details>

```
// Put your logs below this line


```

</details>

### amplifyconfiguration.json

```
{
    "auth": {
        "plugins": {
            "awsCognitoAuthPlugin": {
                "IdentityManager": {
                    "Default": {}
                },
                "CredentialsProvider": {
                    "CognitoIdentity": {
                        "Default": {
                            "PoolId": "[COGNITO IDENTITY POOL ID]",
                            "Region": "[REGION]"
                        }
                    }
                },
                "CognitoUserPool": {
                    "Default": {
                        "PoolId": "[COGNITO USER POOL ID]",
                        "AppClientId": "[COGNITO USER POOL APP CLIENT ID]",
                        "Region": "[REGION]"
                    }
                },
                "Auth": {
                    "Default": {
                        "authenticationFlowType": "USER_SRP_AUTH",
                        "OAuth": {
                            "WebDomain": "[YOUR COGNITO DOMAIN ]",
                            "AppClientId": "[COGNITO USER POOL APP CLIENT ID]",
                            "SignInRedirectURI": "[CUSTOM REDIRECT SCHEME AFTER SIGN IN, e.g. myapp://]",
                            "SignOutRedirectURI": "[CUSTOM REDIRECT SCHEME AFTER SIGN OUT, e.g. myapp://]",
                            "Scopes": [ 
                                "phone",
                                "email",
                                "openid",
                                "profile",
                                "aws.cognito.signin.user.admin"
                            ]
                        }
                    }
                }
            }
        }
    },
"api": {
        "plugins": {
            "awsAPIPlugin": {
                "[API NAME]": {
                    "endpointType": "REST",
                    "endpoint": "[API GATEWAY ENDPOINT]",
                    "region": "[REGION]",
                    "authorizationType": "AWS_IAM",
                    ...
                }
            }
        }
    }
}
```

### GraphQL Schema

<details>

```graphql
// Put your schema below this line


```

</details>

**Comment by joon-won** [comment]
@yktoledo Thanks for the report, we will take a look into this issue

**Comment by joon-won** [comment]
Hey @yktoledo, thank you for your patience. This issue will be addressed with the next release version. Please let us know if there's any further concern.

**Comment by joon-won** [comment]
Closing this issue as the issue has been addressed with our latest release `2.15.1`. Please re-open the issue if you have further question or issue!
